apps/src/templates/sectionProgressV2/ProgressFeedbackBanner.jsx

```diff
@@ -0,0 +1,126 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import {connect} from 'react-redux';
+
+import FeedbackBanner, {
+  BANNER_STATUS,
+} from '@cdo/apps/lib/ui/feedback/FeedbackBanner';
+import i18n from '@cdo/locale';
+
+import {
+  createProgressV2Feedback,
+  fetchProgressV2Feedback,
+} from './progressV2FeedbackRedux';
+
+const FOORM_LINK = 'https://studio.code.org/form/new_progress_feedback';
+
+const ProgressFeedbackBanner = ({
+  canShow,
+  isLoading,
+  progressV2Feedback,
+  fetchProgressV2Feedback,
+  createProgressV2Feedback,
+  errorWhenCreatingOrLoading,
+}) => {
+  const [bannerStatus, setBannerStatus] = React.useState(BANNER_STATUS.UNSET);
+  const [answered, setAnswered] = React.useState(false);
+
+  // Load feedback on mount
+  React.useEffect(() => {
+    fetchProgressV2Feedback();
+  }, [fetchProgressV2Feedback]);
+
+  React.useEffect(() => {
+    // If we can't show the banner, it is unavailable
+    if (!canShow) {
+      setBannerStatus(BANNER_STATUS.UNAVAILABLE);
+      return;
+    }
+
+    // If we are loading or already closed, we shouldn't change states
+    if (isLoading || bannerStatus === BANNER_STATUS.CLOSED) {
+      return;
+    }
+
+    // If we just answered the banner, set it to answered
+    if (answered && bannerStatus === BANNER_STATUS.UNANSWERED) {
+      setBannerStatus(BANNER_STATUS.ANSWERED);
+      return;
+    }
+
+    // If feedback has been loaded and empty and the user hasn't answered, it is unanswered.
+    // If we have feedback and the user did not just submit feedback, close the banner.
+    if (progressV2Feedback && !answered) {
+      if (progressV2Feedback.empty) {
+        setBannerStatus(BANNER_STATUS.UNANSWERED);
+      } else {
+        setBannerStatus(BANNER_STATUS.CLOSED);
+      }
+    }
+  }, [
+    progressV2Feedback,
+    bannerStatus,
+    canShow,
+    isLoading,
+    answered,
+    fetchProgressV2Feedback,
+  ]);
+
+  /**
+   * Effect for handling errors.
+   */
+  React.useEffect(() => {
+    errorWhenCreatingOrLoading && setBannerStatus(BANNER_STATUS.UNSET);
+  }, [errorWhenCreatingOrLoading]);
+
+  const answer = satisfied => {
+    if (bannerStatus === BANNER_STATUS.UNANSWERED) {
+      setAnswered(true);
+      createProgressV2Feedback(satisfied);
+    }
+  };
+
+  const close = () => {
+    setBannerStatus(BANNER_STATUS.CLOSED);
```

Comment:
I wonder if it would be smart to persist this closed state in local storage so that we don't keep showing the banner on every page reload to people who have hit the X in the past. Could be frustrating to keep saying "I don't want to answer this question" and yet having it shown again and again.

Reply:
The `share more` banner is only shown if the user just answered the first question. This is because we only set the state to `ANSWERED` if we have called the `answer` method. Otherwise, if feedback exists we set the state to `CLOSED`

Reply:
Ah, there is no close button on the feedback component before the user has answered the question. This was the designs from LTI, but maybe we should reconsider for our use.